app/assets/javascripts/shared/datatable/main.js

```diff
@@ -200,14 +200,14 @@ class DradisDatatable {
       //
       // To prevent a reset from happening, we just have to ensure that the number of columns on the page
       // matches the length of columns array in the saved state object.
-      stateLoadCallback: function(_settings) {
+      stateLoadCallback: function(_settings, callback) {
         var localStorageData = JSON.parse(localStorage.getItem(that.localStorageKey));
 
-        if (localStorageData !== null) {
-          return that.rebuildSavedStateColumnsFromLocalStorage(localStorageData);
-        } else {
+        if (!localStorageData) {
           return null;
         }
+
+        callback(that.rebuildSavedStateColumnsFromLocalStorage(localStorageData));
```

Comment:
would something like this work instead:
```suggestion
        if (localStorageData) {
          callback(that.rebuildSavedStateColumnsFromLocalStorage(localStorageData));
        }
```

Reply:
We have to return null explicitly. I tried removing it previously but the datatable broke when initializing.

![image](https://user-images.githubusercontent.com/18374402/160360872-480c6f87-fb8e-403f-a88f-2f0a4fea8c59.png)

> If the data is loaded synchronously the return value should be the loaded state (or null if no data was loaded).

https://datatables.net/reference/option/stateLoadCallback